### ISSUE TITLE
 prometheus-adapter running  failt 

### ISSUE THREAD

**Issue by xuezm01** [coreos/prometheus-operator#2116]
exec kubectl  create -f  ./manifests/ ,other  services  running  ,but  prometheus-adapter failt .  log info : 
#kubectl   -n  monitoring  log    prometheus-adapter-74f896d4b9-w9tjb

W1111 10:14:40.462095       1 authentication.go:245] Unable to get configmap/extension-apiserver-authentication in kube-system.  Usually fixed by 'kubectl create rolebinding -n kube-system ROLE_NAME --role=extension-apiserver-authentication-reader --serviceaccount=YOUR_NS:YOUR_SA'
F1111 10:14:40.462139       1 adapter.go:206] unable to install resource metrics API: open /tmp/client-ca-file012996851: no such file or directory

**Comment by brancz** [comment]
cc @metalmatze

**Comment by trustbyte** [comment]
bump, having the same issue

kubectl logs -n monitoring prometheus-adapter-74f896d4b9-nv9ps
```
W1112 10:10:57.491508       1 authentication.go:245] Unable to get configmap/extension-apiserver-authentication in kube-system.  Usually fixed by 'kubectl create rolebinding -n kube-system ROLE_NAME --role=extension-apiserver-authentication-reader --serviceaccount=YOUR_NS:YOUR_SA'

F1112 10:10:57.491558       1 adapter.go:206] unable to install resource metrics API: open /tmp/client-ca-file903981934: no such file or directory
```

**Comment by trustbyte** [comment]
i am trying to figure it out what should i do on my side :)
should i redeploy the hole prometheus on my cluster or just a part of it or..

i deleted the pod, that got redeployed and the error persists so it has to be something else

**Comment by metalmatze** [comment]
Did you make sure that the image is pulled again from quay.io? As the tag didn't change you might need to add `ImagePullPolicy: Always` to pull the latest version. You can remove that afterwards again.
Again, sorry for the inconvenience!

**Comment by trustbyte** [comment]
okey, it is working as expected right now.
for those green (I included, as it was a learning step for me as well) it was like this:

```
$kubectl -n monitoring edit deployment prometheus-adapter -o yaml
```

the line 36 was: imagePullPolicy: IfNotPresent
the same like should be like metalmatze said: imagePullPolicy: Always

save the file and the pod should already be redeployed OK (so it seemed in my case) or just delete the prometheus-adapter-XXXXXX-xxxx pod and that should redeploy with the latest image from quay.io


```
     34       labels:
     35         name: prometheus-adapter
     36     spec:
     37       containers:
     38       - args:
     39         - --cert-dir=/var/run/serving-cert
     40         - --config=/etc/adapter/config.yaml
     41         - --logtostderr=true
     42         - --metrics-relist-interval=1m
     43         - --prometheus-url=http://prometheus-k8s.monitoring.svc:9090/
     44         - --secure-port=6443
     45         image: quay.io/coreos/k8s-prometheus-adapter-amd64:v0.3.0
     46         imagePullPolicy: Always
     47         name: prometheus-adapter
     48         ports:
     49         - containerPort: 6443
```

change back the line 46 to "IfNotPresent" after the pod is running OK.

**Comment by junsionzhang** [comment]
> Did you make sure that the image is pulled again from quay.io? As the tag didn't change you might need to add `ImagePullPolicy: Always` to pull the latest version. You can remove that afterwards again.
> Again, sorry for the inconvenience!
gotcha，thank you . would you mind telling me which version should I use ,the master branch or the laster release?

**Comment by metalmatze** [comment]
When using container images, we recommend using either `quay.io/coreos/k8s-prometheus-adapter-amd64:v0.3.0` or `directxman12/k8s-prometheus-adapter-amd64:v0.3.0` which is the v0.3.0 release.

**Comment by metalmatze** [comment]
I'll close this issue as the root cause has been resolved. Feel free to ask if there are still problem with this.

**Comment by chrisghill** [comment]
Not sure if I should be posting this here or creating a new issue, but I'm getting this same error with prometheus-operator v. 0.27.

I followed the instructions here and changed to `imagePullPolicy: Always` :

`
      labels:
        name: prometheus-adapter
    spec:
      containers:
      - args:
        - --cert-dir=/var/run/serving-cert
        - --config=/etc/adapter/config.yaml
        - --logtostderr=true
        - --metrics-relist-interval=1m
        - --prometheus-url=http://prometheus-k8s.monitoring.svc:9090/
        - --secure-port=6443
        image: quay.io/coreos/k8s-prometheus-adapter-amd64:v0.4.1
        imagePullPolicy: Always
        name: prometheus-adapter
        ports:
        - containerPort: 6443
`

But the pod keeps crashing with the logs:
`
I0207 20:00:22.023838       1 adapter.go:91] successfully using in-cluster auth
W0207 20:00:22.375607       1 authentication.go:245] Unable to get configmap/extension-apiserver-authentication in kube-system.  Usually fixed by 'kubectl create rolebinding -n kube-system ROLE_NAME --role=extension-apiserver-authentication-reader --serviceaccount=YOUR_NS:YOUR_SA'
F0207 20:00:22.375636       1 adapter.go:252] unable to install resource metrics API: configmaps "extension-apiserver-authentication" is forbidden: User "system:serviceaccount:monitoring:prometheus-adapter" cannot get configmaps in the namespace "kube-system"
`

Any suggestions on how to fix it?  I tried changing the image version to 0.3.0 and it still wouldn't work.
